ambry-store/src/main/java/com.github.ambry.store/DiskSpaceAllocator.java

```diff
@@ -212,11 +212,13 @@ void allocate(File destinationFile, long sizeInBytes, String storeId, boolean is
         try {
           Files.move(reserveFile.toPath(), destinationFile.toPath());
         } catch (Exception e) {
-          // return the segment to corresponding store reserve dir or swap reserve dir which it belongs to
-          if (isSwapSegment) {
-            swapReserveFiles.add(sizeInBytes, reserveFile);
-          } else {
-            storeReserveFiles.get(storeId).add(sizeInBytes, reserveFile);
+          // return the segment (if exists) to corresponding store reserve dir or swap reserve dir which it belongs to
+          if (reserveFile.exists()) {
```

Comment:
You are right. This is an extra safe check, although it should not be false. Since prod code rarely execute this if block, I think it's no harm to keep it here.

Reply:
consider adding log.warn in the else case so we can see if there is an issue with this in the future